test/tests/functional/pbs_reservations.py

```diff
@@ -2452,3 +2452,74 @@ def test_shared_exclusive_job_not_in_same_rsv_vnode(self):
         self.assertEqual(
             jid2[1], '(' + vn + '[1]:ncpus=4+' + vn + '[2]:ncpus=4)')
         self.assertNotIn(job1_node, job2_node, errmsg)
+
+    def test_resv_confirm_ignore_set_dedicated_time(self):
+        """
+        Test reservation timing with resv_confirm_ignore set to dedicated_time
+        in sched_config
+        """
+        next_day = datetime.date.today() + datetime.timedelta(days=1)
+        next_day_start = datetime.datetime.combine(
+            next_day, datetime.time(0, 0, 0, 0))
+        next_day_start = PBSLogUtils.convert_date_time(
+            next_day_start.strftime("%m/%d/%Y %H:%M:%S"))
+        next_day_end = next_day_start + 86400
+        self.scheduler.set_sched_config({'resv_confirm_ignore':
+                                         'dedicated_time'})
+
+        self.scheduler.add_dedicated_time(start=next_day_start,
+                                          end=next_day_end)
+        now = time.time()
+        rid = self.submit_reservation(user=TEST_USER, select='1:ncpus=1',
+                                      start=now + 5, end=now + 3600)
+        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
+        self.server.expect(RESV, a, id=rid, offset=5)
+        resv_queue = rid.split('.')[0]
+        a = {'Resource_List.walltime': '00:02:00', ATTR_queue: resv_queue}
+        J1 = Job(attrs=a)
+        jid1 = self.server.submit(J1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        self.server.delete(jid1, wait=True)
+        a = {'Resource_List.walltime': '23:00:00', ATTR_queue: resv_queue}
+        J2 = Job(attrs=a)
+        J2.set_sleep_time(3600)
+        jid2 = self.server.submit(J2)
+        msg = "Job would cross dedicated time boundary"
+        attrs = {ATTR_comment: "Not Running: " + msg, ATTR_state: 'Q'}
```

Comment:
i don't see this getting used (missing an expect?). From what I understand, since you are submitting a reservation that's only 1 hour long but try to submit a job to it which is 23 hours long, the job will not run because it will cross the reservation's boundary itself, right?

Reply:
No, we don't backfill till the end of the reservation.  We run a job because there is a chance its walltime is wrong and it'll finish.  The jobs are going to be deleted at the end of the reservation anyway, so there is no reason to not run it.  The scheduler is smart enough to set the end of the job running in a reservation to the end of the reservation if the end time of the job is past the end of the reservation.  This is actually wrong, the scheduler shouldn't complain about dedicated time since the reservation ends before dedicated time starts.